app/src/main/java/appricottsoftware/clarity/fragments/BrowseFragment.java

```diff
@@ -25,5 +58,81 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     @Override
     public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         // Initialize view lookups, listeners
+        podcasts = new ArrayList<>();
+        channels = new ArrayList<>();
+
+        // Parse browse.json from assets to populate Channel array.
+        parseJSON();
+        populateBrowseGrid();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if(context instanceof PlayerInterface) {
+            playerInterface = (PlayerInterface) context;
+        } else {
+            Log.e(TAG, context.toString() + " must implement PlayerInterface");
+            throw new ClassCastException(context.toString() + " must implement PlayerInterface");
+        }
+    }
+
+    // Parses json file and adds contexts to Channel ArrayList
+    void parseJSON() {
+        try {
+            JSONObject obj = new JSONObject(loadJSONFromAsset());
+            JSONArray resp = obj.getJSONArray("results");
+            for (int i = 0; i < resp.length(); i++) {
+                Channel c = getGson().fromJson(String.valueOf(resp.getJSONObject(i)), Channel.class);
+                c.setUid(ClarityApp.getSession(getActivity()).getUserID());
+                channels.add(c);
+            }
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Gets json from file. Will eventually update this code to pull json from database
+    public String loadJSONFromAsset() {
+        String json = null;
+        try {
+            InputStream is = getActivity().getAssets().open("browse.json");
+            int size = is.available();
+            byte[] buffer = new byte[size];
+            is.read(buffer);
+            is.close();
+            json = new String(buffer, "UTF-8");
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            return null;
+        }
+        return json;
+    }
+
+    private void populateBrowseGrid() {
+        final GridView gridview = getActivity().findViewById(R.id.gv_browse);
+        final ImageTextAdapter adapter = new ImageTextAdapter(getActivity(), channels);
+        gridview.setAdapter(adapter);
+
+        gridview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
```

Comment:
Can you move this onclicklistener into the ViewHolder class? Please see ChannelFragment.java for an example.